components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -1213,16 +1220,16 @@ static void psa_key_management_operation(void)
                 }
 
                 case PSA_CREATE_KEY: {
-                    psa_key_id_t id = 0;
-                    size_t max_bits = 0;
+                    psa_key_id_t id;
+                    id.owner = psa_identity(msg.handle);
 
-                    bytes_read = psa_read(msg.handle, 1, &id, msg.in_size[1]);
+                    bytes_read = psa_read(msg.handle, 1, &(id.key_id), msg.in_size[1]);
```

Comment:
Consider using `PSA_KEY_FILE_GET_KEY_ID` to get the key ID.

Reply:
That wouldn't work here. `PSA_KEY_FILE_GET_KEY_ID` returns the value of `key_id`, here we're reading from the incoming message into `key_id`.

Reply:
`&PSA_KEY_FILE_GET_KEY_ID(id)` wouldn't work?

The downside to how this is written now is it doesn't get the advantage of the macro abstraction and may be depending on more implementation detail than it needs to. This isn't a super strong downside now, but you be aware of it and consider it.

Reply:
My bad, it would work. However I'd like to leave it as is for now, since there's no abstraction for the `owner`.

Reply:
OK